### ISSUE TITLE
Firewall as not paid feature . FR

### ISSUE THREAD

**Issue by DI555** [emanuele-f/PCAPdroid#297]
would you please to make integrated firewall be free feature.
many people use other firewalls, but would skip to newer one..
..so, that will bring much more users than now!!!

as new paid feature would suggest new bunch of colorful themes to choose! not bad for many paid users!!!

thank you in advance!

**Comment by emanuele-f** [comment]
Hi, have you installed the app from Google Play? The app was born has a traffic monitoring app and my approach for monetization is that features not related to its core function should be paid. Are your concerned about the price of the firewall feature?

**Comment by DI555** [comment]
no, not at all, i'm just fan of donation software strategies..,
and thought that if Firewall feature would be a free feature, - more people would leave theirs firewalls and use PCAPdroid!
so, you would have much more users of the app!
and of course some other paid features plus my suggested paid features would bring not less monetization!

**Comment by kevino-techie** [comment]
In my opinion the app already has some great and unique features provided **for free** like the packet capturing and the latest TLS decryption capability which should already attract more people to the App. 
Infact there was no premium offering in the app until 6 months ago (even with tls decryption implemented) and it was being developed for more than 2 years by the dev prior to it. So he best knows how donations worked(s) for him. (Not to mention , the motivation behind developing the app was/is not entirely monetary)

As the developer said, the firewall was not made to be marketed as the core features of the app . There are a bunch of firewall-related apps popping up in the market right now which are bringing different features as per users demands. This will require pcapdroid to shift its focus from its core function to cater such demands. And at one point it would be too much from one single app to offer and for the users to not get confused.

Also in my personal experience, i feel majority of the users won't appreciate the work done on the app ,even when all features are offered for free.
The users who already find the core features useful (like network monitoring /tls decryption) would definitely see the value in using premium firewall feature as well.
So , I currently like this implementation.

**Comment by emanuele-f** [comment]
In part I also agree with @DI555 that providing the firewall for free would attract new users, in particular foss users. Purchasing paid features via the unlock token is not that simple and it relies on Google Play, so such users may try and probably continue to use other firewall apps, which are available for free. Maybe we could provide some basic firewall functions for free (e.g. only blocking of apps), and more advanced as paid (e.g. whitelist mode, ip/domain blocking, autoblock new apps, etc.).

**Comment by kevino-techie** [comment]
> Maybe we could provide some basic firewall functions for free (e.g. only blocking of apps), and more advanced as paid (e.g. whitelist mode, ip/domain blocking, autoblock new apps, etc.).
sounds good. I think netguard also does the same thing.

**Comment by mirh** [comment]
I also agree that the core features are already more than enough unique bang for the (free) buck.
Speaking of the unlock token though.. am I to understand that purchasing that is a sort of "all premium features unlocked" for the google play version too? Because it would seem *natural* for it to be the case automatically, but it's not mentioned.

**Comment by emanuele-f** [comment]
The unlock token is not meant for google play, it's explained in the docs. Many people find it difficult to use it already

**Comment by mirh** [comment]
I know it's not *meant* for the google play version (even though it is purchased through it).
But what I was saying was: suppose that I want to buy ***all*** premium features. Since the *unlock token* purchase already is implicitly a bundle of them all, shouldn't its ownership automatically qualify you?

**Comment by emanuele-f** [comment]
They are two different contexts. The unlock token only exists for non-google-play users
